internal/mode/static/nginx/runtime/manager.go

```diff
@@ -17,18 +17,36 @@ import (
 )
 
 const (
-	pidFile            = "/var/run/nginx/nginx.pid"
+	PidFile            = "/var/run/nginx/nginx.pid"
 	pidFileTimeout     = 10000 * time.Millisecond
-	nginxReloadTimeout = 60000 * time.Millisecond
+	NginxReloadTimeout = 60000 * time.Millisecond
 )
 
 type (
-	readFileFunc  func(string) ([]byte, error)
-	checkFileFunc func(string) (fs.FileInfo, error)
+	ReadFileFunc  func(string) ([]byte, error)
+	CheckFileFunc func(string) (fs.FileInfo, error)
 )
 
 var childProcPathFmt = "/proc/%[1]v/task/%[1]v/children"
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . nginxPlusClient
+
+type nginxPlusClient interface {
+	UpdateHTTPServers(upstream string, servers []ngxclient.UpstreamServer) (added []ngxclient.UpstreamServer, deleted []ngxclient.UpstreamServer, updated []ngxclient.UpstreamServer, err error)
+	GetUpstreams() (*ngxclient.Upstreams, error)
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . ProcessHandler
+
+type ProcessHandler interface {
```

Comment:
I think we can simplify this interface.

```

type ProcessHandler interface {
	FindMainProcess(ctx context.Context, timeout time.Duration) (int, error)
	ReadFile(file string) ([]byte, error)
	Kill(pid int) error
}

type ProcessHandlerImpl struct {
	readFile  ReadFileFunc
	checkFile CheckFileFunc
}

func NewProcessHandlerImpl(readFileFunc ReadFileFunc, checkFileFunc CheckFileFunc) *ProcessHandlerImpl {
	return &ProcessHandlerImpl{readFile: readFileFunc, checkFile: checkFileFunc}
}

func (p *ProcessHandlerImpl) ReadFile(file string) ([]byte, error) {
	return p.readFile(file)
}

func (p *ProcessHandlerImpl) Kill(pid int) error {
	return syscall.Kill(pid, syscall.SIGHUP)
}
...

```

Other changes:
- `FindMainProcess` will use the readFile and checkFile funcs that are stored on ProcessHandlerImpl. 
- `EnsureNginxRunning` method is removed. In `static/manager.go` replace call to `EnsureNginxRunning` with `processHandler.FindMainProcess`.

Reply:
I also don't think ProcessHandler needs to be exported since it's not used outside of this file.

Reply:
I did an update here, but Im not sure that's what you meant @kate-osborn